### DISCUSSION TITLE
Error in set_seed() function || AttributeError: module 'torch._C' has no attribute '_has_mps'

### DISCUSSION THREAD

**Discussion by saish23** [mrdbourke/pytorch-deep-learning#674] [Q&A]
Hi, I am getting this error below because of which I am unable to run the whole code. Error is encountered when I am calling this set_seed() a function that is copied below. Kindly help me with this. Thanks in advance.

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-9-e029fb818f73>](https://localhost:8080/#) in <cell line: 7>()
      5 # Since we're creating a new layer with random weights (torch.nn.Linear),
      6 # let's set the seeds
----> 7 set_seeds()
      8 
      9 # Update the classifier head to suit our problem

2 frames
[/usr/local/lib/python3.10/dist-packages/torch/mps/__init__.py](https://localhost:8080/#) in manual_seed(seed)
     50     # sure mps is available (otherwise we just return without
     51     # erroring out)
---> 52     if not torch._C._has_mps:
     53         return
     54     seed = int(seed)

AttributeError: module 'torch._C' has no attribute '_has_mps'

**Comment by mrdbourke** [comment]
Hi @saish23 , 

Can you please show me the code of `set_seeds()`? 

Is it the same code as the course?

```python
def set_seeds(seed: int=42):
    """Sets random sets for torch operations.

    Args:
        seed (int, optional): Random seed to set. Defaults to 42.
    """
    # Set the seed for general torch operations
    torch.manual_seed(seed)
    # Set the seed for CUDA torch operations (ones that happen on the GPU)
    torch.cuda.manual_seed(seed)
```
